File: dataset/Intermediate/Translation/Avatar/v1-Python-code/codeforces_31_A.py
from sklearn.utils import shuffle

def my_decorator(func):
    shuffle([99, 49, 21])

    def dec_result(*args, **kwargs):
        result = func(*args, **kwargs)
        return result
    return dec_result
from scipy.stats import ttest_ind
ttest_ind([55, 100, 83], [88, 83, 60])
input_number = int(input())
arr = [int(i) for i in input().split()]
for i in range(len(arr)):
    for second_index in range(len(arr)):
        if i == second_index:
            continue

        @my_decorator
        def recursive_check(k, stop, step):
            if step == 0 or (step > 0 and k >= stop) or (step < 0 and k <= stop):
                return
            if arr[k] != arr[second_index] and arr[k] != arr[i] and (arr[k] == arr[i] + arr[second_index]):
                print(k + 1, second_index + 1, i + 1)
                exit()
            recursive_check(k + step, stop, step)
        recursive_check(0, len(arr), 1)
print(-1)

File: scripts/GAs/Experiment_Results_v1-Python-code/intermediate/SR/Translation/use_test/deepseek-coder-6.7b-instruct/Avatar/Python/Java-sanitized/codeforces_31_A.java
import java.util.*;

public class codeforces_31_A {
    private static int[] arr;
    private static int input_number;

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        input_number = scanner.nextInt();
        arr = new int[input_number];
        for (int i = 0; i < input_number; i++) {
            arr[i] = scanner.nextInt();
        }
        for (int i = 0; i < arr.length; i++) {
            for (int second_index = 0; second_index < arr.length; second_index++) {
                if (i == second_index)
                    continue;
                recursiveCheck(0, arr.length, 1, i, second_index);
            }
        }
        System.out.println(-1);
    }

    private static void recursiveCheck(int k, int stop, int step, int i, int second_index) {
        if (step == 0 || (step > 0 && k >= stop) || (step < 0 && k <= stop))
            return;
        if (arr[k] != arr[second_index] && arr[k] != arr[i] && (arr[k] == arr[i] + arr[second_index])) {
            System.out.println((k + 1) + " " + (second_index + 1) + " " + (i + 1));
            System.exit(0);
        }
        recursiveCheck(k + step, stop, step, i, second_index);
    }
}